OpenRA.Game/MiniYaml.cs

```diff
@@ -129,26 +154,29 @@ public MiniYaml Clone()
 				}
 				catch (ArgumentException ex)
 				{
-					throw new InvalidDataException($"Duplicate key '{y.Key}' in {y.Location}", ex);
+					throw new InvalidDataException("Duplicate key '{0}' in {1}".F(y.Key, y.Location), ex);
 				}
 			}
 
 			return ret;
 		}
 
-		public MiniYaml(string value)
-			: this(value, null) { }
-
-		public MiniYaml(string value, List<MiniYamlNode> nodes)
+		MiniYaml FindNodeValueByKey(string key)
 		{
-			Value = value;
-			Nodes = nodes ?? new List<MiniYamlNode>();
+			if (key == null)
+				return null;
+
+			foreach (var y in Nodes)
+				if (key.Equals(y.Key))
+					return y.Value;
+
+			return null;
 		}
 
 		public static List<MiniYamlNode> NodesOrEmpty(MiniYaml y, string s)
```

Comment:
This can then be removed, changing the two places that call it to use `.TryGetValue`.